packages/ember-metal/lib/property_get.js

```diff
@@ -61,7 +61,11 @@ export function get(obj, keyName) {
   if (isDescriptor) {
     return value.get(obj, keyName);
   } else if (isPath(keyName)) {
```

Comment:
I think `isPath(keyName) && !(keyName in obj)` would be enough here

Reply:
Good catch 👍